router/voteCallbacks.go

```diff
@@ -257,13 +283,114 @@ func (r *Router) getEnvelopeList(request routerRequest) {
+		r.sendError(request, fmt.Sprintf("cannot get envelope list: (%s)", err))
+		return
+	}
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getValidatorList(request routerRequest) {
+	var response types.MetaResponse
+	var err error
+	response.ValidatorList, err = r.vocapp.State.Validators(true)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get validator list: %v", err))
+		return
+	}
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getBlock(request routerRequest) {
+	var response types.MetaResponse
+	if request.Height > r.vocapp.Height() {
+		r.sendError(request, fmt.Sprintf("block height %d not valid for vochain with height %d", request.Height, r.vocapp.Height()))
+		return
+	}
+	response.Block = types.BlockMetadataFromBlockModel(r.Scrutinizer.App.GetBlockByHeight(int64(request.Height)))
+	if response.Block == nil {
+		r.sendError(request, fmt.Sprintf("cannot get block: no block with height %d", request.Height))
+		return
+	}
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getBlockByHash(request routerRequest) {
+	var response types.MetaResponse
+	response.Block = types.BlockMetadataFromBlockModel(r.Scrutinizer.App.GetBlockByHash(request.Payload))
+	if response.Block == nil {
+		r.sendError(request, fmt.Sprintf("cannot get block: no block with hash %x", request.Payload))
+		return
+	}
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getBlockList(request routerRequest) {
+	var response types.MetaResponse
+	for i := 0; i < request.ListSize; i++ {
+		if int64(request.From)+int64(i) > r.vocinfo.Height() {
+			break
+		}
+		response.BlockList = append(response.BlockList, types.BlockMetadataFromBlockModel(r.Scrutinizer.App.GetBlockByHeight(int64(request.From)+int64(i))))
+	}
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getTx(request routerRequest) {
+	var response types.MetaResponse
+	tx, hash, err := r.Scrutinizer.App.GetTxHash(request.Height, request.TxIndex)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get tx: %v", err))
+		return
+	}
+	response.Tx = &types.TxPackage{
+		Tx:          tx.Tx,
+		BlockHeight: request.Height,
+		Index:       request.TxIndex,
+		Hash:        hash,
+		Signature:   tx.Signature,
+	}
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getTxListForBlock(request routerRequest) {
+	var response types.MetaResponse
+	block := r.vocapp.Node.BlockStore().LoadBlock(int64(request.Height))
+	if block == nil {
+		r.sendError(request, fmt.Sprintf("cannot get tx list: block does not exist"))
+		return
+	}
+	if request.ListSize > MaxListSize || request.ListSize <= 0 {
+		request.ListSize = MaxListSize
+	}
+	maxIndex := request.From + request.ListSize
+	for i := request.From; i < maxIndex && i < len(block.Txs); i++ {
+		signedTx := new(models.SignedTx)
+		tx := new(models.Tx)
+		proto.Unmarshal(block.Txs[i], signedTx)
+		proto.Unmarshal(signedTx.Tx, tx)
```

Comment:
these can error, you should check for it

Reply:
or, if you're sure this error can never happen:

```
if err := proto.Unmarshal(...); err != nil {
    panic(err) // should never happen, because <reason>
}
```